docs/hypervisors.md

```diff
@@ -0,0 +1,48 @@
+# Hypervisors
+
+* [Hypervisors](#hypervisors)
+    * [Types](#types)
+    * [Determine currently configured hypervisor](#determine-currently-configured-hypervisor)
+    * [Choose a Hypervisor](#choose-a-hypervisor)
+
+## Types
+
+Kata Containers supports multiple hypervisors. Since they offer different
+features and options, Kata Containers provides a separate
+[configuration file](/src/runtime/README.md#configuration)
+for each.
+
+> **Note:**
+>
+> The simplest way to switch between hypervisors is to create a symbolic link
+> to the appropriate hypervisor-specific configuration file.
+
+| Hypervisor | Written in | Architectures | Type | Comment | Configuration file |
+|-|-|-|-|-|-|
+[ACRN](https://projectacrn.org/) | C | `x86_64` | Type 1 | "bare metal" | `configuration-acrn.toml` |
+[Cloud Hypervisor](https://github.com/cloud-hypervisor/cloud-hypervisor) | rust | `aarch64`, `x86_64` | Type 2 | [KVM] | `configuration-clh.toml` |
+[Firecracker](https://github.com/firecracker-microvm/firecracker) | rust | `aarch64`, `x86_64` | Type 2 | [KVM] | `configuration-fc.toml` |
+[QEMU](http://www.qemu-project.org/) | C | all | Type 2 | [KVM] | `configuration-qemu.toml` |
+
+[KVM]: https://en.wikipedia.org/wiki/Kernel-based_Virtual_Machine
+
+## Determine currently configured hypervisor
+
+```bash
+$ kata-runtime kata-env | awk -v RS= '/\[Hypervisor\]/' | grep Path
+```
+
+## Choose a Hypervisor
+
+The table below provides a brief summary of some of the differences between
+the hypervisors, allowing you to decide which are most appropriate for
+you:
+
+| Hypervisor | Summary | Features | Limitations | Container Creation speed | Memory density | Use cases | Comment |
```

Comment:
You are taking a risk here, because you are using "subjective" terms.

On the other hand, this is extremely valuable in understanding the trade-offs.

What about instead showing what the focus is, something like putting the use-case columns first

ACRN: Safety critical and real-time workloads, like embedded and IOT systems
CLH: Low latency, small memory footprint and small attack surface for high-performance modern cloud workloads
Firecracker: Very slim, ideal for serverless or FaaS use cases
QEMU: Full-featured hypervisor, suitable for a majority of users

If you want to have container creation speed and memory usage information, that's fine, but I would then put actual typical numbers rather than vague "good / better / best" in order to help someone really evaluate what that means.

Reply:
Yes, they are subjective terms. But we cannot put in numbers realistically - there are just too many factors so any numbers would be effectively meaningless. For that reason I've tried to suggest the relative pros/cons of each hypervisor, but, as always, YMMV :smile:

In summary: this was written as a very high level document which just introduces some of the options user have. It is not a whitepaper and the document is not written to allow a full evaluation - the wording was chosen to provide an indication of the hypervisors users may wish to explore further. I've added a brief introduction to hint at this.